src/WebJobs.Extensions.Http/ClaimsIdentityHelper.cs

```
@@ -0,0 +1,94 @@
+using System.Security.Claims;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Http
+{
+    /// <summary>
+    /// Provides methods to grab the ClaimsPrincipal from an HTTP Request trigger
+    /// </summary>
+    public static class ClaimsIdentityHelper
+    {
+        // Shared serializer instance. This is safe for multi-threaded use.
+        private static readonly DataContractJsonSerializer Serializer = GetCustomSerializer();
+
+        /// <summary>
+        /// Retrieves a serialized ClaimsIdentity object from the http request of an HTTP Trigger
+        /// </summary>
+        /// <param name="request">The request message from the HTTP Trigger</param>
+        /// <param name="identityHeaderName">The name of the header the identity is stored on.</param>
+        /// <returns></returns>
+        public static ClaimsIdentity GetIdentityFromHttpRequest(HttpRequest request, string identityHeaderName)
+        {
+            if (request == null)
+            {
+                return null;
+            }
+
+            string claimsIdentityHeaderValue = GetClaimsIdentityHeaderValue(request, identityHeaderName);
+            if (claimsIdentityHeaderValue == null)
+            {
+                return null;
+            }
+            return FromBase64EncodedJson(claimsIdentityHeaderValue);
+        }
+
+        /// <summary>
+        /// Adds a serialized ClaimsIdentity object to a specified http header
+        /// </summary>
+        /// <param name="request">The HTTP request to add the header to</param>
+        /// <param name="identity">The ClaimsIdentity to be serialized</param>
+        /// <param name="headerName">The name of the header to add the serialized value to</param>
+        public static void AddIdentityToHttpRequest(HttpRequest request, ClaimsIdentity identity, string headerName)
+        {
+            if (request == null)
+            {
+                return;
+            }
```

Same comment about throwing instead of returning null here.